third_party/move/move-compiler-v2/tests/checking/access_specifiers/access_ok.move

```diff
@@ -0,0 +1,50 @@
+module 0x42::m {
+
+    struct S has store {}
+    struct R has store {}
+    struct T has store {}
+    struct G<T> has store {}
+
+    fun f1() acquires S {
+    }
+
+    fun f2() reads S {
+    }
+
+    fun f3() writes S {
+    }
+
+    fun f4() acquires S(*) {
+    }
+
+    fun f_multiple() acquires R reads R writes T, S reads G<u64> {
```

Comment:
The existing acquires notation uses comma, so this would be a breaking change. I think the use comma was inspired by Rust, which also often uses comma where other languages have semicolon.

Reply:
@wrwg note that I was talking about using the semicolon to separate each access specifier (not the multiple resources that can be specified within each access specifier), not just replacing `,` with `;`. Given that currently we can specify the `acquires` keyword just once per function signature (and use `,` to specify multiple resources), I am not sure I understand why what I suggested is a breaking change.

Reply:
Ah yes, you are right, before there was only one acquires possible. Stil it would not fit Move syntax style. Semicolon is nowhere used besides in function bodies. We separate abilities with comma (`store, key`) and field declarations are separated with comma as well.

Reply:
In at least one place, we separate abilities with `+`, not `,`: https://aptos.dev/move/book/generics#declaring-constraints, when specifying constraints on a type parameter (I assume we had to do this to avoid parsing conflicts, i.e., distinguish an additional type parameter vs. additional ability, but I wish we had used the `+` syntax for the `has` clause as well to be consistent).

I don't know if `+` would fit Move syntax style? That is,

`fun f_multiple() acquires R, T + reads R, G<u8> + writes T, S + reads G<u64> {}`

Or other options like `and`, `&&`?
`fun f_multiple() acquires R, T and reads R, G<u8> and writes T, S and reads G<u64> {}`

My main point is, we should discuss considering some sort of separator between access specifiers. Otherwise, when multiple resource specifiers and multiple access specifiers are present, the block of text is hard to mentally parse: it is also an isolated case (as far as I can think right now) where multiple "clauses" are separated purely with white space.

I should note specifically that this is not a blocker/action item for this PR, just a point of discussion as and when this issue occurred to me. We can move it offline/elsewhere if this is not the appropriate forum.

Reply:
Certainly we can continue to discuss this. I have little concern about human recognition, because a style also the pretty printer would enforce, is put them on multiple lines:

```
fun foo()
  reads A 
  writes B
```

Reply:
Fair enough (about using formatters to resolve this issue).